sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -191,6 +191,128 @@ public void processElement(InputT element) throws Exception {
     processTimestampedElement(TimestampedValue.atMinimumTimestamp(element));
   }
 
+  /** @deprecated Use {@link TestPipeline} with the {@code DirectRunner}. */
+  @Deprecated
+  public void processRetraction(InputT element) throws Exception {
```

Comment:
Do not bother changing `DoFnTester`. Just make anything that is required a stub that crashes.